app/src/main/java/io/pslab/communication/ScienceLab.java

```diff
@@ -446,52 +446,58 @@ public int[] oscilloscopeProgress() {
     }
 
     private boolean fetchData(int channelNumber) {
-        int samples = this.aChannels.get(channelNumber - 1).length;
```

Comment:
Some formatting issues there.
It'd be great if you could format your files after you finish work and before you commit them.
Just select the whole file (`Ctrl+A`) and then run reformat (`Ctrl+Alt+L`).